core/server/common/src/main/java/alluxio/conf/ServerConfiguration.java

```diff
@@ -154,22 +117,17 @@ public static void set(PropertyKey key, Object value) {
    * @param source the source of the the properties (e.g., system property, default and etc)
    */
   public static void set(PropertyKey key, Object value, Source source) {
-    Preconditions.checkArgument(key != null && value != null && !value.equals(""),
-        String.format("The key value pair (%s, %s) cannot be null", key, value));
-    Preconditions.checkArgument(!value.equals(""),
-        String.format("The key \"%s\" cannot be have an empty string as a value. Use "
-            + "Configuration.unset to remove a key from the configuration.", key));
-    PROPERTIES.put(key, String.valueOf(value), source);
+    sConf.set(key, value, source);
   }
 
   /**
    * Unsets the value for the appropriate key in the {@link Properties}.
    *
    * @param key the key to unset
    */
-  public static void unset(PropertyKey key) {
+  public static void unset(@Nonnull PropertyKey key) {
```

Comment:
we expect everything to be `@Nonnull` by default, so we don't explicitly annotate it